### PR TITLE
feat: apis for workflow endtime and status

### PR DESCRIPTION
**Pull request by STRRL**
Signed-off-by: STRRL <str_ruiling@outlook.com>

### What problem does this PR solve?

Add two fields in workflow dashboard API:
- end time, the time of workflow accomplished
- status, the status of workflow, available values: `Running`, `Succeed`, `Failed`, `Unknown` 

### What is changed and how does it work?
We use conditions called `Accomplished` and `Scheduled` to describe the status of workflow
- `Scheduled` means the entry node has been spawned, the workflow is running
- `Accomplished` means the entry nodes has been accomplished, the workflow is succeed

The status which returned in dashboard API is calculated by these conditions.

> Note: we have not resolved the failure explicitly, which will be finished in other PR.

Also, I have updated and refactored the workflow entry reconciler, just like previous refactors on serail/chaos nodes: 
- perform operation without depending on status
- then update the status 

### Checklist

Tests

- [ ] Unit test
- [ ] E2E test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Breaking backward compatibility

Related changes

- [ ] Need to update the documentation

### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

**Comment by STRRL** [comment]
I think this PR is ready for review , PTAL

File: pkg/workflow/controllers/workflow_entry_reconciler.go [hunk 1]
```diff
@@ -46,51 +59,191 @@ func (it *WorkflowEntryReconciler) Reconcile(request reconcile.Request) (reconci
...
-			if err != nil {
-				it.logger.Error(err, "failed to update workflowNeedUpdate status")
-				return err
+			if WorkflowNodeFinished(entryNodes[0].Status) {
+				SetWorkflowCondition(&workflowNeedUpdate.Status, v1alpha1.WorkflowCondition{
+					Type:   v1alpha1.WorkflowConditionAccomplished,
+					Status: corev1.ConditionTrue,
+					Reason: "",
+				})
+				if workflowNeedUpdate.Status.EndTime != nil {
```
Comment (g1eny0ung) [inline pkg/workflow/controllers/workflow_entry_reconciler.go:156]: `== nil` I think?

**Review by g1eny0ung — COMMENTED** [review]
The rest LGTM.

File: pkg/workflow/controllers/workflow_entry_reconciler.go (hunk 1 above)
Comment (STRRL) [reply pkg/workflow/controllers/workflow_entry_reconciler.go]: fixed in 35c07ed

**Review by g1eny0ung — APPROVED** [review]

**Review by fewdan — APPROVED** [review]
LGTM

**Pull request merged**